Objects/structseq.c

```diff
@@ -365,9 +366,99 @@ structseq_reduce(PyStructSequence* self, PyObject *Py_UNUSED(ignored))
     return NULL;
 }
 
+
+static PyObject *
+structseq_replace(PyStructSequence *self, PyObject *args, PyObject *kwargs)
+{
+    PyObject *tup = NULL;
+    PyObject *dict = NULL;
+    PyObject *result;
+    Py_ssize_t n_fields, n_visible_fields, n_unnamed_fields, i;
+
+    if (!_PyArg_NoPositional("__replace__", args)) {
+        return NULL;
+    }
+
+    n_fields = REAL_SIZE(self);
+    if (n_fields < 0) {
+        return NULL;
+    }
+    n_visible_fields = VISIBLE_SIZE(self);
+    n_unnamed_fields = UNNAMED_FIELDS(self);
+    if (n_unnamed_fields < 0) {
+        return NULL;
+    }
+    tup = _PyTuple_FromArray(self->ob_item, n_visible_fields);
+    if (!tup) {
+        goto error;
+    }
+
+    dict = PyDict_New();
+    if (!dict) {
+        goto error;
+    }
+
+    if (kwargs != NULL) {
+        for (i = 0; i < n_visible_fields; i++) {
+            const char *name = Py_TYPE(self)->tp_members[i].name;
+            PyObject *key = PyUnicode_FromString(name);
+            if (!key) {
+                goto error;
+            }
+            PyObject *ob = _PyDict_Pop(kwargs, key, self->ob_item[i]);
+            Py_DECREF(key);
+            if (!ob) {
+                goto error;
+            }
+            PyTuple_SetItem(tup, i, ob);
+        }
+        for (i = n_visible_fields; i < n_fields; i++) {
+            const char *name = Py_TYPE(self)->tp_members[i - n_unnamed_fields].name;
+            PyObject *key = PyUnicode_FromString(name);
+            if (!key) {
+                goto error;
+            }
+            PyObject *ob = _PyDict_Pop(kwargs, key, self->ob_item[i]);
+            if (PyDict_SetItem(dict, key, ob) < 0) {
+                goto error;
+            }
+        }
+        if (PyDict_Size(kwargs) > 0) {
+            PyObject *names = PyDict_Keys(kwargs);
+            if (names) {
+                PyErr_Format(PyExc_ValueError, "Got unexpected field name(s): %R", names);
```

Comment:
I think that TypeError is more appropriate exception.

Yes, `namedtuple` raises ValueError, but it is unusual. Perhaps it should be changed.

Reply:
It is only for InitVar fields. TypeError is raised for unknown names.

Reply:
Thanks for the clarification. I update the exception type to `TypeError` for both `PyStructSequence` and `collections.namedtuple`.